src/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -324,7 +324,17 @@ public void focusGained(final FocusEvent e) {
     m_rightHandSidePanel.add(m_smallView, BorderLayout.NORTH);
     m_tabsPanel.setBorder(null);
     m_rightHandSidePanel.add(m_tabsPanel, BorderLayout.CENTER);
-    m_actionButtons = new ActionButtons(m_data, m_mapPanel, this);
+
+    MovePanel movePanel = new MovePanel(m_data, m_mapPanel, this);
+    m_actionButtons = new ActionButtons(m_data, m_mapPanel, movePanel, this);
+
+    List<KeyListener> keyListeners = ImmutableList.of(this.getArrowKeyListener(), movePanel.getUndoMoveKeyListener());
+    for( KeyListener keyListener : keyListeners ) {
+      m_mapPanel.addKeyListener(keyListener );
+      // TODO: figure out if it is really needed to double add the key listener to both the frame and also the map panel
+      this.addKeyListener(keyListener);
```

Comment:
Not really consistency, motivation is more because it's unclear when and which one is unnecessary. The behavior was still buggy when there was only one listener, though that was before I had a slightly better understanding of the remove listener bug #305 (which also impacts the 'n' key as well), regardless I still do not fully understand the bug in #305, nor which of the listeners are actually fully needed.

The changes for the 'u' key were meant to be similar to those as the 'n' key. It  was an unpleasant surprise to see this double add.